endpoints/osp/osp

```diff
@@ -0,0 +1,582 @@
+#!/usr/bin/env bash
+# -*- mode: sh; indent-tabs-mode: nil; sh-basic-offset: 4 -*-
+# vim: autoindent tabstop=4 shiftwidth=4 expandtab softtabstop=4 filetype=bash
+
+# This script implements the 'osp' endpoint for rickshaw.  It runs 1 or more
+# clients and servers as instances/VMs in an Openstack cluster.
+#
+# Usage:
+# osp [--validate] --endpoint-opts=host=<host>,user=<user>,client:n-m,o-p,server:n-m,o-p 
+#                    --run-id <id> --base-run-dir --image <location>
+#                    --roadblock-server <host> --roadblock-id <id> --roadblock-passwd=<passwd>
+#
+# If --validate is used all options after client/server will be ignored
+#
+
+# Source the base file for common functions and config
+this_endpoint_dir=$(dirname `readlink -e $0` | sed -e 'sX/binXX')
+endpoint_base_dir=$(cd $this_endpoint_dir >/dev/null && cd .. && /bin/pwd)
+if [ -e "$endpoint_base_dir/base" ]; then
+    . "$endpoint_base_dir/base"
+else
+    echo "Could not find endpoint source file "$endpoint_base_dir/base", exiting"
+    exit 1
+fi
+
+echo "#running osp endpoint"
+clear_all_servers="1"
+chroot_rbind_mounts="proc dev sys lib/modules usr/src boot var/run"
+image_cache_size=3
+default_flavor="nfv-large"
+default_availability_zone="nova"
+osruntime="podman"
+host_mounts=""
+endpoint_name="osp"
+instance_prefix="rickshaw"
+base_osruntime_bootstrap="$endpoint_base_dir/osruntime-bootstrap"
+project_name="crucible-rickshaw"
+new_osp_followers="" # engines which only run tools for openstack nodes
+vm_image="centos9"
+vm_user="centos"
+declare -A cpuPartitioning
+declare -A flavor
+declare -A availability_zone
+declare -A ips
+declare -A active_compute_nodes
```

Comment:
Since osruntime is not declared here does that mean it can't be changed per engine?

Reply:
yes